### ISSUE TITLE
Enhancement: templateselect to use template names instead of file names

### ISSUE THREAD

**Issue by JeromeBreizhtorm** [bolt/bolt#3338]
The dropdown for a templateselect field is populated with file names. It would be more user-friendly to use template names.

Chopping the file extension, replacing underscores and ucfirst-ing the filename could be a start, even if it would be even better to have an association name/filename somewhere, like this in contenttypes.yml for example :

```
template:
    type: templateselect
    templates:
        greattemplate:
            name:'A great template'
            filename:'great-template.twig'
        othertemplate:
            name:'Anoter template'
            filename:'bad_naming.twig'
```

**Comment by Pinpickle** [comment]
The theme config would be the best place for defining template names, I think. This would do well to be paired with the templatefields definition.

**Comment by JeromeBreizhtorm** [comment]
Of course ... would be way better in the theme config. +1

**Comment by GawainLynch** [comment]
Addressed in #3447
